src/snowflake/snowpark/mock/_plan.py

```diff
@@ -388,6 +397,32 @@ def handle_function_expression(
     return _MOCK_FUNCTION_IMPLEMENTATION_MAP[func_name](*to_pass_args)
 
 
+def handle_udf_expression(
+    exp: FunctionExpression,
+    input_data: Union[TableEmulator, ColumnEmulator],
+    analyzer: "MockAnalyzer",
+    expr_to_alias: Optional[Dict[str, str]],
+    current_row=None,
+):
+    udf = analyzer.session.udf._registry.get(exp.udf_name)
+
+    if udf is None:
+        raise SnowparkSQLException(
+            f"[Local Testing] udf {exp.udf_name} does not exist."
+        )
+
+    to_pass_args = [
+        calculate_expression(child, input_data, analyzer, expr_to_alias).name
+        for child in exp.children
+    ]
+
+    res = input_data.apply(lambda row: udf(*[row[col] for col in to_pass_args]), axis=1)
+    res.sf_type = ColumnType(exp.datatype, exp.nullable)
```

Comment:
Is `exp.datatype`, `exp.nullable` filled here?

Reply:
I also have the question that is exp the same as the udf definition (what is returned as columns)?